### PR TITLE
Influxdb udp-client usage added

### PR DESCRIPTION
**Pull request by mguryanov**

### CODE REVIEW

File: config.yml.example [hunk 1]
```diff
@@ -4,11 +4,22 @@ pinba:
...
+# Http usage:
+# "addr" should be of the form "http://host:port"
+# or "http://[ipv6-host%zone]:port".
+#
   addr: "http://influxdb:8086"
   user: "username"
   password: "secret"
   database: "pinba-database"
 
+# Udp usage:
```
Comment (olegfedoseev) [inline config.yml.example:17]: Why it's after block, and not with http usage example? 
For me, it makes more sense to put it together as a comment for `addr` field.

File: main.go [hunk 2]
```diff
@@ -45,6 +50,22 @@ func getConfig(filename string) (*config, error) {
...
 
+func newInfluxdbClient(cfg *config, userAgent string) (influxdbClient, error) {
+	if strings.HasPrefix(cfg.Influxdb.Addr, "http") {
+		return influxdb.NewHTTPClient(influxdb.HTTPConfig{
+			Addr:      cfg.Influxdb.Addr,
+			Username:  cfg.Influxdb.User,
+			Password:  cfg.Influxdb.Password,
+			UserAgent: userAgent,
+		})
+	} else {
```
Comment (olegfedoseev) [inline main.go]: `else` is redundant, because we have `return`

File: main.go [hunk 3]
```diff
@@ -5,13 +5,18 @@ import (
...
 	"time"
 
 	influxdb "github.com/influxdata/influxdb/client/v2"
 	pinba "github.com/olegfedoseev/pinba-server/client"
 	yaml "gopkg.in/yaml.v2"
 )
 
+type influxdbClient interface {
+	Write(influxdb.BatchPoints) error
+}
```
Comment (olegfedoseev) [inline main.go]: What is the point of this interface if we not using it? Why not return concrete struct type?

**Review by olegfedoseev — CHANGES_REQUESTED** [review]

File: config.yml.example (hunk 1 above)
Comment (mguryanov) [reply config.yml.example:17]: udp usage has slightly different config requirements
fields like user, password and database not required
that is why this emphasized in separate block

File: main.go (hunk 2 above)
Comment (mguryanov) [reply main.go]: yes, you are right
i will fix it

File: main.go (hunk 3 above)
Comment (mguryanov) [reply main.go]: i think it's design matter - introducing some cheap abstraction for simplify main flow understanding
and we are using this interface in function
```
func newInfluxdbClient(cfg *config, userAgent string) (influxdbClient, error)
```

**Comment by olegfedoseev** [comment]
Let's split changes into two PRs - one for UDP client, I'll merge it, and one with "packaging", and here I have doubts. I'm not sure I want Arch specific Dockerfile and PKGBUILD in here.

File: main.go (hunk 3 above)
Comment (olegfedoseev) [reply main.go]: It has to have some goal, like testability or reusability or at least readability.
Here I see none of the above.
And if you want to use interface, why not original one https://github.com/influxdata/influxdb/blob/master/client/v2/client.go#L66

Comment (mguryanov) [reply main.go]: Ok!
using original interface :)
I'm will create new PR from udp-client branch, where fixed it

**Pull request closed without merging**